apps/sage_intacct/utils.py

```
@@ -252,6 +253,60 @@ def sync_employees(self):
             employee_attributes, self.workspace_id)
         return account_attributes
 
+    
+    def post_employees(self, employee: ExpenseAttribute, auto_map_employee_preference: str):
+        """
+        Create a Vendor on Sage Intacct
+        :param auto_map_employee_preference: Preference while doing automap of employees
+        :param vendor: vendor attribute to be created
+        :return Vendor Destination Attribute
+        """
+
+        general_mappings = GeneralMapping.objects.get(workspace_id=employee.workspace_id)
+
+        sage_intacct_display_name = employee.detail['employee_code'] if (
+            auto_map_employee_preference == 'EMPLOYEE_CODE' and employee.detail['employee_code']
+        ) else employee.detail['full_name']
+
+        name = employee.detail['full_name']
+
+        try:
+            contact = {
+                'CONTACTNAME': sage_intacct_display_name,
+                'PRINTAS': sage_intacct_display_name,
+                'EMAIL1': employee.value,
+                'FIRSTNAME': name[0],
+                'LASTNAME': name[-1] if len(name) == 2 else None
+            }
+
+            created_contact = self.connection.contacts.post(contact)
+        
+        except Exception as e:
+            logger.error(exception.response)
```

why are we silently ignoring this error? When does this happen? Also, you are trying to create employee even if contact creation is failed.

---

contact creation will fail if the contact is already present so we are raising that exception.

---

discussed;